ray-operator/controllers/ray/raycluster_controller_test.go

```diff
@@ -289,6 +289,34 @@ var _ = Context("Inside the default namespace", func() {
 				listResourceFunc(ctx, &workerPods, workerFilterLabels, &client.ListOptions{Namespace: "default"}),
 				time.Second*2, time.Millisecond*200).Should(Equal(4), fmt.Sprintf("workerGroup %v", workerPods.Items))
 		})
+
+		It("should delete all head and worker pods if suspended", func() {
+			// suspend a Raycluster and check that all pods are deleted.
+			err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				Eventually(
+					getResourceFunc(ctx, client.ObjectKey{Name: myRayCluster.Name, Namespace: "default"}, myRayCluster),
+					time.Second*9, time.Millisecond*500).Should(BeNil(), "My raycluster = %v", myRayCluster)
+				suspend := true
+				myRayCluster.Spec.Suspend = &suspend
+				return k8sClient.Update(ctx, myRayCluster)
+			})
+			Expect(err).NotTo(HaveOccurred(), "failed to update test RayCluster resource")
+
+			// check that all pods are deleted
+			Eventually(
+				listResourceFunc(ctx, &workerPods, workerFilterLabels, &client.ListOptions{Namespace: "default"}),
+				time.Second*15, time.Millisecond*500).Should(Equal(0), fmt.Sprintf("workerGroup %v", workerPods.Items))
+
+			Eventually(
+				listResourceFunc(ctx, &headPods, headFilterLabels, &client.ListOptions{Namespace: "default"}),
+				time.Second*15, time.Millisecond*500).Should(Equal(0), fmt.Sprintf("head %v", headPods.Items))
+		})
+
+		It("cluster's .status.state should be updated to 'suspended' shortly after all Pods are terminated", func() {
+			Eventually(
+				getClusterState(ctx, "default", myRayCluster.Name),
+				time.Second*(common.RAYCLUSTER_DEFAULT_REQUEUE_SECONDS+5), time.Millisecond*500).Should(Equal(rayv1.Suspended))
+		})
```

Comment:
> Perhaps we need to add a test for updating Spec.Suspend from true to false.

I originally had this but it got a little messy because it required updating every pod's status subresource back to "Running" state.. I can add it back if we think the coverage is important.

Reply:
I added a test case for setting `suspend` back to `false` and checking pods are re-created and `status.state` updated to `ready`